### PR TITLE
Handle celery errors by re-running async task

### PR DESCRIPTION
**Pull request by demianbrecht**

### CODE REVIEW

File: django_declarative_apis/machinery/tasks.py [hunk 1]
```diff
@@ -183,7 +184,22 @@ def schedule_future_task_runner(task_runner_args, task_runner_kwargs,
...
+        # Once in a while, a connection in the pool will timeout prior to a health check being called in redis-py and
+        # will result in an error being raised here. This should be removed once the issue has been sorted out.
+        # Note: This is around the use of redis-py in celery where celery's event loop is not running
+        # https://github.com/celery/kombu/issues/1019
+        try:
+            _run()
+            return
+        except kombu.exceptions.OperationalError as err:
+            logger.warn('kombu.exceptions.OperationalError')
+            pass
```
Comment (smholloway) [inline django_declarative_apis/machinery/tasks.py]: Should we log how many attempts we're at and if we hit 3?

File: django_declarative_apis/machinery/tasks.py [hunk 2]
```diff
@@ -183,7 +184,21 @@ def schedule_future_task_runner(task_runner_args, task_runner_kwargs,
...
+        # will result in an error being raised here. This should be removed once the issue has been sorted out.
+        # Note: This is around the use of redis-py in celery where celery's event loop is not running
+        # https://github.com/celery/kombu/issues/1019
+        try:
+            future_task_runner.apply_async(task_runner_args, task_runner_kwargs, queue=queue, routing_key=routing_key, countdown=countdown+delay)
+            return
+        except kombu.exceptions.OperationalError as err:
+            logger.warn('kombu.exceptions.OperationalError (attempt: %s)', attempt)
+            if attempt >= MAX_ATTEMPTS - 1:
+                raise err
```
Comment (dshafer) [inline django_declarative_apis/machinery/tasks.py]: I'd like to have the option to fall back to running the task synchronously, perhaps controlled by a django setting, or else an option in the `DeferrableTask` decorator

**Comment by dshafer** [comment]
I like `DECLARATIVE_ENDPOINT_FORCE_SYNCHRONOUS_TASKS`, but what I'm looking for is a bit different - I think it might be useful to have an option to run tasks synchronously _only if_ we fail to run them in the queue.  So we try to do the right thing (queue/async), but fall back to synchronous processing if the queue goes wrong for some reason.

**Comment by demianbrecht** [comment]
@dshafer so demanding ;) Should be satisfied by the most recent commit.

**Pull request merged**